File: src/main/java/io/vertx/ext/consul/v1/agent/Check.java
package io.vertx.ext.consul.v1.agent;

import io.vertx.core.AsyncResult;
import io.vertx.core.Handler;

public interface Check {

  Check register(RegisterPayload payload, Handler<AsyncResult<Boolean>> handler);

  Check deregister(String checkId, Handler<AsyncResult<Boolean>> handler);

  Check pass(String checkId, String note, Handler<AsyncResult<Boolean>> handler);

  Check warn(String checkId, String note, Handler<AsyncResult<Boolean>> handler);

  Check fail(String checkId, String note, Handler<AsyncResult<Boolean>> handler);

  Check update(String checkId, String status, String output, Handler<AsyncResult<Boolean>> handler);
}

File: vertx-consul-client/src/main/kotlin/io/vertx/kotlin/ext/consul/Check.kt
package io.vertx.kotlin.ext.consul

import io.vertx.ext.consul.Check
import io.vertx.ext.consul.CheckStatus

fun Check(
    id: String? = null,
  name: String? = null,
  nodeName: String? = null,
  notes: String? = null,
  output: String? = null,
  serviceId: String? = null,
  serviceName: String? = null,
  status: CheckStatus? = null): Check = io.vertx.ext.consul.Check().apply {

  if (id != null) {
    this.id = id
  }

  if (name != null) {
    this.name = name
  }

  if (nodeName != null) {
    this.nodeName = nodeName
  }

  if (notes != null) {
    this.notes = notes
  }

  if (output != null) {
    this.output = output
  }

  if (serviceId != null) {
    this.serviceId = serviceId
  }

  if (serviceName != null) {
    this.serviceName = serviceName
  }

  if (status != null) {
    this.status = status
  }

}

